src/main/java/io/vertx/core/net/impl/KeyStoreHelper.java

```diff
@@ -137,12 +135,20 @@ public static KeyStoreHelper create(VertxInternal vertx, TrustOptions options) t
   private final KeyStore store;
   private final Map<String, X509KeyManager> wildcardMgrMap = new HashMap<>();
   private final Map<String, X509KeyManager> mgrMap = new HashMap<>();
+  private final Map<String, KeyStore> trustMgrMap = new HashMap<>();
 
   public KeyStoreHelper(KeyStore ks, String password) throws Exception {
     Enumeration<String> en = ks.aliases();
     while (en.hasMoreElements()) {
       String alias = en.nextElement();
       Certificate cert = ks.getCertificate(alias);
+      if (ks.isCertificateEntry(alias) && ! alias.startsWith(DUMMY_CERT_ALIAS)){
+        KeyStore keyStore = KeyStore.getInstance("jks");
+        keyStore.load(null, null);
+        keyStore.setCertificateEntry("cert-1", cert);
+        trustMgrMap.put(alias, keyStore);
+      }
+      // TODO this should only be done if: ks.isKeyEntry(alias)
```

Comment:
I've added this TODO to start the discussion with @vietj, it's off topic regarding the core feature of this PR, but I think the sni name gathering for server certificates shouldn't be done for CA certificate entries. What do you think.

Reply:
can you elaborate ?

Reply:
If the KeyStore is created from JKSOptions with a user provided TrustStore the store could contain CertificateEntries and KeyEntries at once. The loop currently also adds entries to the mgrMap for CNs from CertificateEntries.

Reply:
I don't get it : the JKSOptions are different fields for keystore (`keyCertOptions`) and truststore (`trustOptions`) on `TCPSSLOptions`

Reply:
A vert.x user creates a Java key store with a CertificateEntry and and a KeyEntry like this pseudo code and stores it in "myKeyStore" file:

```
 keyStore = KeyStore.getInstance("jks");
 keyStore.load(null, null);
 keyStore.setCertificateEntry("trustedCA", caCertBuf);
 PrivateKey key = privateKey;
 Certificate[] chain = certChain;
 keyStore.setEntry("keyCert", new KeyStore.PrivateKeyEntry(key, chain), new KeyStore.PasswordProtection("dummy".toCharArray()));
```

and uses this key store for TrustOptions and KeyCertOptions:

```
    NetServerOptions netServerOptions = new NetServerOptions().setTrustOptions(new JksOptions().setPath("myKeyStore"))
        .setKeyCertOptions(new JksOptions().setPath("myKeyStore"));
    vertx.createNetServer(netServerOptions);
```

Reply:
ah ok but in reality those can be two different files, that was confusing

Reply:
you can implement the todo condition imho, it's better to do it.

Reply:
done